### PR TITLE
fix: Prompt shop without rewards when a wild Pokemon flees

### LINKED ISSUE #6230: [Balance] Put the player in the shop without rewards when a wild Pokemon flees.
Right now, if a wild pokemon flees the battle due to an effect like teleport or emergency exit, the game will progress to the next wave without having the player enter the reward screen.

PRs like #6197 have been opened to address this. After conferring with balance team, the desired behavior is for situations like this to cause the player to be taken to the end of battle screen, except without the ability to choose rewards.

The desired behavior is as follows:
- Applies whenever a wild encounter ends with the opposing pokemon fleeing / being forced out
   - This includes both player-sourced and enemy-sourced fleeing effects: whirlwind, dragon tail, teleport, emergency exit, etc.
   - This does **not** apply if the player chooses to run from the battle -- this should retain existing behavior where the player proceeds directly to the next battle
- Take the player to the reward screen with the ability to do all of the things they normally could do on battle end, except be given a choice of rewards -- just like in those mystery encounters where the player can "continue".
- For double battles, at least one pokemon must have fainted and not fled to be shown shop rewards (diverging from current behavior with golisopod and emergency exit).

### PR DESCRIPTION
**Pull request by miguelc-29**
Fixes #6230 

## What are the changes the user will see?
When a wild Pokémon ends the encounter by fleeing or being forced out (e.g., using Teleport, Emergency Exit, or Roar/Whirlwind), the player will now be taken to the post-battle shop screen instead of immediately advancing to the next wave. 
- If no opposing Pokémon fainted, the shop UI will appear with no reward choices (allowing the player to heal/manage their party).
- If at least one opposing Pokémon fainted (e.g., in a double battle), partial/normal rewards will be shown based on the number of defeated enemies.
- Choosing the "Run" option manually preserves the existing behavior and skips the reward screen.

## Why am I making these changes?
This change aligns the game mechanics with official Pokémon behavior. Previously, wild flee/force-out scenarios were skipping the reward screen entirely. This resulted in an inconsistent user experience, preventing players from accessing the shop to heal or buy items after a battle ended through means other than fainting or catching. This PR standardizes the end-of-battle flow for non-victory force-outs while preserving intentional skips (like standard running away).

### 1. Core Logic (`src/phases/battle-end-phase.ts`)
The primary logic changes are contained within the battle end phase to intercept encounters that end prematurely without a victory flag:
* **State Interception:** Added logic to check if a non-victory wild battle has concluded without a modifier selection phase already being queued in the state machine.
* **Dynamic Reward Generation:**
    * **Partial/Full Defeat (`faintedCount > 0`):** If at least one opposing Pokémon fainted (crucial for Double Battles), the phase transitions into the standard modifier selection, populating rewards based on the `faintedCount`.
    * **Zero Defeats (`faintedCount === 0`):** If the enemy fled or was forced out with no faints, the game queues a modifier selection screen explicitly configured as a "continue only" mystery encounter, which provides the shop UI but generates zero loot.
* **Preserving Intentional Skips:** Explicit checks ensure that if the battle ended due to the player manually selecting the "Run" command, this new logic is bypassed, preserving the legacy behavior of skipping the shop entirely.

### 2. Automated Tests
* **Move Mechanics (`roar.test.ts`, `teleport.test.ts`, `metronome.test.ts`):** Added tests to verify that player-initiated force-outs (Roar/Whirlwind) and enemy-initiated escapes (Teleport) correctly trigger the empty shop state.
* **Ability Mechanics (`emergency-exit.test.ts`, `wimp-out.test.ts`, `honey-gather.test.ts`):** Created  coverage for `Emergency Exit` to ensure threshold-triggered flee effects correctly route the player to the post-battle shop. Minor adjustments were made to `Wimp Out` and `Honey Gather` tests to accommodate the state machine adjustments.
* **Battle State Edge Cases (`battle.test.ts`, `double-battle.test.ts`):** Added specific test cases to validate the complex scenario where one enemy is defeated but the other flees, ensuring the reward generation correctly scales to `faintedCount === 1` rather than skipping rewards or granting full rewards.

## Screenshots/Videos
https://youtu.be/-d6RaVyLyro

## How to test the changes?
1. Start a wild encounter.
2. Force the enemy out using a move like Roar or Whirlwind, or wait for an enemy with Teleport to use it.
3. Verify that the post-battle shop screen appears without reward options.
4. In a double battle, defeat one enemy and force the other out. Verify that the shop appears with rewards corresponding to one defeated enemy.
5. Manually choose "Run" and verify it correctly skips the shop.

## Checklist

  - [x] I'm using `beta` as my base branch
  - [x] The current branch is not named `beta`, `main` or the name of another long-lived feature branch
  - [x] I have provided a clear explanation of the changes within the PR description
  - [x] The PR title matches the Conventional Commits format (as described in [CONTRIBUTING.md](https://github.com/pagefaultgames/pokerogue/blob/beta/CONTRIBUTING.md#pr-title-format))
- [x] The PR is self-contained and cannot be split into smaller PRs
- [x] There is no overlap with another open PR
- [x] I have tested the changes manually
- [x] The full automated test suite still passes (use `pnpm test:silent` to test locally)
- [x] I have created new automated tests (`pnpm test:create`) or updated existing tests related to the PR's changes if necessary
- [x] I have provided screenshots/videos of the changes (if applicable)

### CODE REVIEW

**Review by SirzBenjie — DISMISSED** [review]

**Review by damocleas — DISMISSED** [review]

File: test/tests/battle/double-battle.test.ts [hunk 1]
```diff
@@ -90,6 +105,44 @@ describe("Double Battles", () => {
...
+
+    game.move.select(MoveId.ROAR, 0, 3);
+    game.move.select(MoveId.SPLASH, 1);
+
+    const handler = await getModifierShopHandler();
+    expect(handler.options.length).toBe(0);
+  });
+
+  it("should offer partial rewards when one opponent is defeated and the other flees", async () => {
+    game.override.battleStyle("double");
```
Comment (fabske0) [inline test/tests/battle/double-battle.test.ts:128]: What is partial rewards? The changes to BattleEndPhase only check if at least 1 enemy fainted and not how many

File: test/tests/battle/battle.test.ts [hunk 2]
```diff
@@ -181,4 +183,19 @@ describe("Phase - Battle Phase", () => {
...
+
+    vi.spyOn(activeOverrides, "RUN_SUCCESS_OVERRIDE", "get").mockReturnValue(true);
+
+    const commandPhase = game.scene.phaseManager.getCurrentPhase() as CommandPhase;
+    commandPhase.handleCommand(Command.RUN, 0);
+
+    await game.toNextWave();
+
+    expect(game.scene.currentBattle.waveIndex).toBe(2);
+    expect(game.scene.ui.getMode()).not.toBe(UiMode.MODIFIER_SELECT);
```
Comment (fabske0) [inline test/tests/battle/battle.test.ts]: Running away doesn't skip the modifierSelectPhase. The test just passes because you are going directly to the next wave

File: test/tests/battle/double-battle.test.ts (hunk 1 above)
Comment (miguelc-29) [reply test/tests/battle/double-battle.test.ts:128]: Sorry, it was a mistake, rewards are normal if one of the opponent is defeated, that is the behavior I think its expected and is happening. I will change the description of the test.

File: test/tests/battle/battle.test.ts (hunk 2 above)
Comment (miguelc-29) [reply test/tests/battle/battle.test.ts]: I fixed it, the changes:
- added flag to track successful run in battle.ts
- in the check in battle-end-phase if it is a run doesn't trigger the selectmodifierphase
- fixed the tests to match behavior

**Review by Wlowscha — APPROVED** [review]

**Review by SirzBenjie — APPROVED** [review]
